### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by chanchal16**
- Created basic authentication using mockbee
-  Login page with guest credentials
-  Signup page

https://user-images.githubusercontent.com/56486295/163665341-5d780946-1414-49f6-80fa-da69341e687d.mp4

### CODE REVIEW

File: src/services/auth.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import axios from "axios";
+
+const SignUpUser = (name,email,password)=>{
+    return axios.post('/api/auth/signup',{
```
Comment (sonishreyas) [inline src/services/auth.js:4]: you shoiuld add error handling here

File: src/App.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import "./App.css";
 import { Navbar,SideNav } from "./components/common";
-import { VideoListing } from "./pages";
+import { Routes, Route } from "react-router-dom";
+import {Login,Signup,VideoListing } from "./pages";
 
 function App() {
```
Comment (ravikumar0403) [inline src/App.js:6]: You can change this to ES6 syntax

File: src/contexts/MainProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import {AuthContextProvider} from './AuthContext';
+import {VideoContextProvider} from './VideoContext';
+
+function MainContextProvider({children}) {
+  return (
+    <div>
```
Comment (ravikumar0403) [inline src/contexts/MainProvider.jsx:6]: `div` is not required here. It will just add an unnecessary in DOM

**Review by ravikumar0403 — APPROVED** [review]

**Review by sonishreyas — APPROVED** [review]
The code looks good. You can also implement RequiresAuth functionality here. Rest looks good

File: src/services/auth.js (hunk 1 above)
Comment (chanchal16) [reply src/services/auth.js:4]: > you shoiuld add error handling here

I have done it in context only and I don't think there will be any problem but if required I refactor it later.

**Pull request merged**